### ISSUE TITLE
TypeError: cannot use a string pattern on a bytes-like object when converting to PDF

### ISSUE THREAD

**Issue by khink** [dottedmag/archmage#27]
First off, thanks for maintaining this project. Thanks to this, i've been able to convert a .chm to HTML, works perfectly!

So `$ archmage -x my_file.chm` works fine. But when i try to convert to PDF `$ archmage -c pdf my_file.chm`, i get:
```
Traceback (most recent call last):
  File "/home/kees/Projects/archmage/env/bin/archmage", line 33, in <module>
    sys.exit(load_entry_point('archmage', 'console_scripts', 'archmage')())
  File "/home/kees/Projects/archmage/archmage/cli.py", line 192, in main
    source.htmldoc(options.output, options.mode)
  File "/home/kees/Projects/archmage/archmage/CHM.py", line 402, in htmldoc
    self.extract_entries(
  File "/home/kees/Projects/archmage/archmage/CHM.py", line 352, in extract_entries
    self.extract_entry(
  File "/home/kees/Projects/archmage/archmage/CHM.py", line 327, in extract_entry
    Entry(
  File "/home/kees/Projects/archmage/archmage/CHM.py", line 496, in correct
    data = re.sub("<div .*teamlib\\.gif.*\\/div>", "", data)
  File "/usr/lib/python3.8/re.py", line 210, in sub
    return _compile(pattern, flags).sub(repl, string, count)
TypeError: cannot use a string pattern on a bytes-like object
```

This looks like the kind of error that is the result of an incomplete conversion to Python 3. (I've also installed archmage in a Python 2 virtualenv, but that fails on a print statement: `print(msg, file=outfp)`.)

I've installed archmage from source, commit 8b7f3cde46f998205910d398344469a1e7a255c9. Python is 3.8.5. I also have `Successfully installed archmage beautifulsoup4-4.9.3 pychm-0.8.6 sgmllib3k-1.0.0 soupsieve-2.1`

Once again, thanks for your work maintaining this, and if i need to provide more info, let me know!

**Comment by dottedmag** [comment]
I see the problem, will have a look into the cause soon.

**Comment by prosper00** [comment]
I have the same thing with -c html (and -c pdf, as noted)

**Comment by Sembiance** [comment]
I took a look into this a bit. Changed all the `re.sub("whatever")` calls to  `re.sub(b"whatever")` and that fixed those errors.  But there were more errors in htmldoc such as with the `str.join(input, "\n")` call. Reversing the args fixed the error, but it didn't run htmldoc correctly. I ended up giving up here and just stuck with the 'extract' mode for now.

Looks like maybe this module was once python2 and not everything was brought up to python3
